[docs] Reduce the use of "our", "We"... (#17347)
mui-org/material-ui: docs/src/pages/guides/typescript/typescript.md

This is because TypeScript [widens the return types of function expressions](https://github.com/Microsoft/TypeScript/issues/241).

> Because of this, we recommend using the `createStyles` helper function to construct your style rules object:

Author: ```suggestion
Because of this, using the `createStyles` helper function to construct your style rules object is recommended:
```

how they want to narrow that type down. We do not offer the possibility to use a generic
type in those cases for [the same reasons `event.target` is not generic in React](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/11508#issuecomment-256045682).

The demos include typed variants that use type casting. It is an acceptable tradeoff
because the types are all located in a single file and are very basic. You have to decide for yourself

> if the same tradeoff is acceptable for you. We want the library types to be strict

Author: ```suggestion
if the same tradeoff is acceptable for you. The library types are be strict
```
